### PR TITLE
WebEx Notification Plugin

### PR DESCRIPTION
**Pull request by bastell**

### CODE REVIEW

**Comment by jeaubin** [comment]
When you get a notification it would be nice to know how to view a log

Either trade log if its internal or the path of the zip with pyats logs view <zip>

File: src/pyats/contrib/plugins/webex.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
+import logging
+import os
+import json
+import requests
+from pyats.easypy.plugins.bases import BasePlugin
+from pyats import configuration as cfg
+
+logger = logging.getLogger("pyats.contrib.webexnotify")
```
Comment (simingy) [inline src/pyats/contrib/plugins/webex.py:8]:
```python
logger = logging.getLogger(__name__)
```
would do

File: src/pyats/contrib/plugins/webex.py [hunk 2]
```diff
@@ -94,7 +98,46 @@ def post_job(self, job):
...
+            host = job.runtime.env['host']['name']
+            # Determine if liveview is running
+            if self.runtime.args.liveview and\
+                    self.runtime.args.liveview_keepalive:
+                # Liveview will set this to the assigned port if not specified
+                port = self.runtime.args.liveview_port
+                try:
+                    # Attempt to add a link using the host domain name
+                    addr = socket.getfqdn()
+                    socket.gethostbyname(addr)
```
Comment (jeaubin) [inline src/pyats/contrib/plugins/webex.py:122]: What's the point of this line?

Comment (bastell) [reply src/pyats/contrib/plugins/webex.py:122]: This is a check that the address is resolvable. `socket.getfqdn()` will return the hostname even if that isn't a domain on the network.

File: src/pyats/contrib/plugins/webex.py [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+Aborted       : {job.results[aborted]}
+Blocked       : {job.results[blocked]}
+Skipped       : {job.results[skipped]}
+Errored       : {job.results[errored]}
+```
+"""
+
+class WebExTeamsNotifyPlugin(BasePlugin):
+    '''
+    Runs after job, sends notification with results to a specified WebEx Teams
```
Comment (tahigash) [inline src/pyats/contrib/plugins/webex.py:38]: no notification when starting job? if the job in queue on some systems, better to inform when starting the job.

File: src/pyats/contrib/plugins/webex.py [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+        if space:
+            payload['roomId'] = space
+        elif email:
+            payload['toPersonEmail'] = email
+
+        logger.info('Sending WebEx Teams notification')
+
+        try:
+            # Attempt POST
+            r = requests.post(MESSAGE_URL,
```
Comment (tahigash) [inline src/pyats/contrib/plugins/webex.py:150]: This is REST. why not develop rest.connector for webex teams and call it here? with the rest.connector, we can have device in testbed yaml and also we can send notification even during runs and in testcases and anytime when user wants.

**Pull request merged**